clash-vexriscv/src/VexRiscv.hs

```diff
@@ -259,23 +260,39 @@ vexRiscv# !_sourcePath clk rst0
   jtag_TDI = unsafePerformIO $ do
     (v, initStage1, initStage2, stepRising, stepFalling, _shutDown) <- vexCPU
 
+    -- Make sure all the inputs are defined
+    let
+      rst0' = unsafePerformIO . makeDefinedRandom <$> unsafeToActiveHigh rst0
```

Comment:
Could we move `unsafePerformIO . makeDefinedRandom` to a separate function called `unsafeMakeDefinedRandom`? It's generally encouraged to put `unsafePerformIO` calls into a separate function with a `NOINLINE` annotation to prevent weird compiler optimizations.